include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1402,22 +1452,10 @@ namespace internal
 
         // ... coarse mesh (needed since user vector might be const)
         {
-          IndexSet locally_relevant_dofs;
-
-          if (mg_level_coarse == numbers::invalid_unsigned_int)
-            DoFTools::extract_locally_relevant_dofs(dof_handler_coarse,
-                                                    locally_relevant_dofs);
-          else
-            DoFTools::extract_locally_relevant_level_dofs(
-              dof_handler_coarse, mg_level_coarse, locally_relevant_dofs);
```

Comment:
This is a bit too conservative. I have changed it to active DoFs plus constraining DoFs.